internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -487,3 +497,32 @@ func (r *BundleReconciler) updateStatus(ctx context.Context, orig *fleet.Bundle,
 	metrics.BundleCollector.Collect(ctx, bundle)
 	return nil
 }
+
+// cleanupOrphanedBundleDeployments will delete all existing BundleDeployments corresponding that do not match a provided list
+func (r *BundleReconciler) cleanupOrphanedBundleDeployments(ctx context.Context, bundle *fleet.Bundle, created []*fleet.BundleDeployment) error {
+	uidsToKeep := make(sets.Set[types.UID], len(created))
+	for _, bd := range created {
+		uidsToKeep.Insert(bd.UID)
+	}
```

Comment:
If we only use `created` for this purpose, wouldn't it make more sense to simply feed the equivalent of `uidsToKeep` directly to `cleanupOrphanedBundleDeployments`?
L282 would then append the `bd.UID` to a slice of `types.UID` (aka `string`), and we'd save ourselves some complexity.

Reply:
Thanks, good point. I applied this suggestion but kept using a `Set`, as it works better for lookups.